src/ngraph_utils.h

```diff
@@ -205,6 +205,9 @@ void NgraphSerialize(const std::string&,
 // Collect the total memory usage through /proc/self/stat
 void MemoryProfile(long&, long&);
 
+// Get the inputs, outputs, and depdencies for the node
```

Comment:
This function is very specific to nGraph Node - so it should really be added to nGraph so that other parts of nGraph (bridge/backend) can make use it.

NGraph Node already has a `<<` operator and a function called write_long_description() - which you should update with printing information you are printing here. 
https://github.com/NervanaSystems/ngraph/blob/aa52081c2e5acac6dd343d2c16cc3a00b7676dab/src/ngraph/node.hpp#L152

Reply:
I removed this node logging function from the bridge now and will add it to nGraph core.